src/negotiation/maker.ts

```diff
@@ -0,0 +1,28 @@
+import { ExecutionParams, Order } from "./payload";
+
+export class Maker {
```

Comment:
This naming feels weird to me.
Right now it indicates to me that some logic is behind the implementation (a market maker comes up with a rate, etc.). 
I think you tried to separate out some concerns but I'm not sure you made it better.

Reply:
Terminologies have changed, please check.